core/market/decentralized/src/matcher.rs

```diff
@@ -140,60 +181,334 @@ impl Matcher {
         demand_id: &SubscriptionId,
         id: &Identity,
     ) -> Result<(), MatcherError> {
-        Ok(self.store.remove_demand(demand_id, id).await?)
+        self.store.remove_demand(demand_id, id).await?;
+
+        log::info!(
+            "Unsubscribed Demand: [{}] using identity: {} [{}]",
+            &demand_id,
+            id.name,
+            id.identity
+        );
+        Ok(())
+    }
+
+    pub async fn list_our_offers(&self) -> Result<Vec<Offer>, QueryOffersError> {
+        let identities = self.identity.list().await?;
+        let store = self.store.clone();
+
+        let mut our_offers = vec![];
+        for node_id in identities.into_iter() {
+            our_offers.append(&mut store.get_offers(Some(node_id)).await?)
+        }
+
+        Ok(our_offers)
+    }
+
+    pub async fn list_our_unsubscribed_offers(
+        &self,
+    ) -> Result<Vec<SubscriptionId>, QueryOffersError> {
+        let identities = self.identity.list().await?;
+        let store = self.store.clone();
+
+        let mut our_offers = vec![];
+        for node_id in identities.into_iter() {
+            our_offers.append(&mut store.get_unsubscribed_offers(Some(node_id)).await?)
+        }
+
+        Ok(our_offers)
     }
 }
 
-pub(crate) async fn on_offer_received(
+// =========================================== //
+// Discovery protocol messages handlers
+// =========================================== //
+
+pub(crate) async fn on_offer_ids_received(
```

Comment:
misleading name.  Below is my proposition. more descriptive one
```suggestion
pub(crate) async fn filter_known_offer_ids(
```